tests/filecheck/transforms/memref_to_snax.mlir

```diff
@@ -58,3 +76,38 @@
 // CHECK-NEXT:   %22 = "snax.alloc"(%21) <{"memory_space" = 1 : i32}> : (index) -> !llvm.struct<(!llvm.ptr, !llvm.ptr, i32, !llvm.array<2 x i32>, !llvm.array<2 x i32>)>
 // CHECK-NEXT:   %23 = "builtin.unrealized_conversion_cast"(%22) : (!llvm.struct<(!llvm.ptr, !llvm.ptr, i32, !llvm.array<2 x i32>, !llvm.array<2 x i32>)>) -> memref<8x8xi32, #tsl.tsl<[2, 4] -> (16, 4), [2, 4] -> (128, 32)>, 1 : i32>
 // CHECK-NEXT: }) : () -> ()
+
+// -----
+
+"builtin.module"() ({
+  %0 = "test.op"() : () -> (index)
+  %1 = "memref.alloc"(%0) <{"alignment" = 64 : i64, "operandSegmentSizes" = array<i32: 1, 0>}> : (index) -> memref<8x?xi32, #tsl.tsl<[2, 4] -> (16, 4), [?, 4] -> (128, 32)>, 1 : i32>
```

Comment:
The TSL documentation also mentions
> The layout provided allows for some flexibility in defining dynamic shapes within a matrix:
`#tsl.tsl<[?, 4] -> (32, 4), [?, 4] -> (?, 1)>`

Dumb comment: Here you are only testing with dynamic tile size, should you also add a test here for the dynamic stride case?